kernel/objects/capability-spinning/objects/CapEntry.cc

```diff
@@ -125,28 +137,41 @@ namespace mythos {
     return true;
   }
 
-  optional<void> CapEntry::unlink()
+  bool CapEntry::kill(Cap expected)
   {
-    auto next = Link(_next).withoutFlags();
-    auto prev = Link(_prev).withoutFlags();
-    next->_prev.store(prev.value());
-    prev->_next.store(next.value());
-    _prev.store(Link().value());
+    CapValue expectedValue = expected.value();
```

Comment:
It can't be implemented in the Capability as `Cap` is an immutable value that gets passed around. Here, we are changing the capability in the mutable entry (it's container).

Reply:
ah okay, the std::atomic<Capability> is in CapEntry, not inside the Capability. Still the whole function looks strange, maybe a more precise name could help? It is trying to kill/zombify the capability if it is still what we expect. `try_kill`?